Display rotate

schmofu opened the pull request:
Added possibility to rotate the display to the left. The joystick rotates as well.
I am insure about the implications of changes to the config file. So i reused the option "lcdmirrored" which ist currently out of use and not really needed anymore.
Setting lcdmirrored to 1 will result in the desired rotation.
(There ist also a rotation test in the testapp.)

schmofu committed:
changes in testapp, to test and try things.

schmofu committed:
feldtest now has an exit page

schmofu committed:
new test for display rotation

schmofu committed:
Added display left rotation support

If you want to wear the badge in portrait mode instead of landscape, the display and the joystick should turn accordingly. This patch uses the unused config option mirrored to achieve this.
Set lcdmirrored=1 to switch to portrait mode.

schmofu committed:
Fixed display not beeing rotated after boot

In camp main.c the display init was called after config was applied. Thus the setting got overwritten with the default display rotation.
Also unsetting the option didn't work properly.

Sec42 commented:
In 2755f1369708ff7dbd458ff9555579241a1e57bd you moved lcd_Init to before readConfig, this breaks the vdd-fix for affected badges.

Can you please fix this.

Also please keep style-changes like the top of fc964658f3c1d16b27d8b16562f03a41503a312c out of feature patches.

Thanks

schmofu committed:
Corrections as suggested

Set functions now named accordingly. Should now play nicely with original campapp/main.c
